test/infrastructure/vcsim/controllers/vspherevm_controller.go

```diff
@@ -0,0 +1,306 @@
+	WatchFilterValue string
+}
+
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=vspherevms,verbs=get;list;watch;update;patch;delete
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=vsphereclusters,verbs=get;list;watch
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=vspheremachines,verbs=get;list;watch
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=vsphereclusteridentities,verbs=get;list;watch
+// +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=machines,verbs=get;list;watch
+// +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=clusters,verbs=get;list;watch
+// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
+
+// Reconcile ensures the back-end state reflects the Kubernetes resource state intent.
+func (r *VSphereVMReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Fetch the VSphereVM instance
+	vSphereVM := &infrav1.VSphereVM{}
+	if err := r.Client.Get(ctx, req.NamespacedName, vSphereVM); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	// Fetch the owner VSphereMachine.
+	vSphereMachine, err := util.GetOwnerVSphereMachine(ctx, r.Client, vSphereVM.ObjectMeta)
+	// vsphereMachine can be nil in cases where custom mover other than clusterctl
+	// moves the resources without ownerreferences set
+	// in that case nil vsphereMachine can cause panic and CrashLoopBackOff the pod
+	// preventing vspheremachine_controller from setting the ownerref
+	if err != nil || vSphereMachine == nil {
+		log.Info("Owner VSphereMachine not found, won't reconcile")
+		return reconcile.Result{}, err
+	}
+	log = log.WithValues("VSphereMachine", klog.KObj(vSphereMachine))
+	ctx = ctrl.LoggerInto(ctx, log)
+
+	// Fetch the Machine.
+	machine, err := capiutil.GetOwnerMachine(ctx, r.Client, vSphereMachine.ObjectMeta)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if machine == nil {
+		log.Info("Waiting for Machine Controller to set OwnerRef on VSphereMachine")
+		return ctrl.Result{}, nil
+	}
+	log = log.WithValues("Machine", klog.KObj(machine))
+	ctx = ctrl.LoggerInto(ctx, log)
+
+	// Fetch the Cluster.
+	cluster, err := capiutil.GetClusterFromMetadata(ctx, r.Client, machine.ObjectMeta)
+	if err != nil {
+		log.Info("VSphereVM owner Machine is missing cluster label or cluster does not exist")
+		return ctrl.Result{}, err
+	}
+	if cluster == nil {
+		log.Info(fmt.Sprintf("Please associate this machine with a cluster using the label %s: <name of cluster>", clusterv1.ClusterNameLabel))
+		return ctrl.Result{}, nil
+	}
+	log = log.WithValues("Cluster", klog.KObj(cluster))
+	ctx = ctrl.LoggerInto(ctx, log)
+
+	// Return early if the object or Cluster is paused.
+	// NOTE: we are using vSphereMachine because VSphereVM is not a valid v1.Object
+	if annotations.IsPaused(cluster, vSphereMachine) {
```

Comment:
```suggestion
	if annotations.IsPaused(cluster, vSphereVM) {
```
Shouldn't this work?

Maybe I'm misinterpreting the comment

Reply:
Fixed, nice catch